rosdyn_core/include/rosdyn_core/primitives.h

```diff
@@ -559,6 +600,72 @@ inline void Joint::fromUrdf(const urdf::JointPtr& urdf_joint, const rosdyn::Link
   computeJacobian();
 }
 
+
+inline int Joint::enforceLimitsFromRobotDescriptionParam(const std::string& full_param_path, std::string& what)
+{
+  //=============================================================
+  if(!ros::param::has(full_param_path))
+  {
+    what = "Parameter '" + full_param_path + "' is not in rosparam server.\n";
+    return -1;
+  }
+
+  std::string joint_limits_param = full_param_path +"/joint_limits/" + m_name;
+  if(!ros::param::has(joint_limits_param))
+  {
+    what = "Parameter '" + joint_limits_param + "' is not in rosparam server.\n";
+    return -1;
+  }
+  //=============================================================
+
+  try
+  {
+    bool has_velocity_limits;
+    if(!ros::param::get(joint_limits_param + "/has_velocity_limits", has_velocity_limits))
+    {
+      has_velocity_limits = false;
+    }
+    bool has_acceleration_limits;
+    if (!ros::param::get(joint_limits_param +  "/has_acceleration_limits", has_acceleration_limits))
+    {
+      has_acceleration_limits = false;
+    }
+    if (has_velocity_limits)
+    {
+      double vel;
+      if(!ros::param::get(joint_limits_param + "/max_velocity", vel))
+      {
+        what += (what.length()>0 ? "\n" : "")
+             + joint_limits_param + "/max_velocity is not defined. URDF value is superimposed";
+      }
+      else
+      {
+        m_Dq_max = vel >0 ? vel : m_Dq_max;
+      }
+    }
+
+    if (has_acceleration_limits)
+    {
+      double acc;
```

Comment:
The acceleration is not set to default value

Reply:
Done!